app/src/org/commcare/provider/IdentityCalloutHandler.java

```diff
@@ -0,0 +1,174 @@
+package org.commcare.provider;
+
+import android.content.Intent;
+
+import org.commcare.android.javarosa.IntentCallout;
+import org.commcare.commcaresupportlibrary.identity.IdentityResponseBuilder;
+import org.commcare.commcaresupportlibrary.identity.model.IdentificationMatch;
+import org.commcare.commcaresupportlibrary.identity.model.MatchResult;
+import org.commcare.commcaresupportlibrary.identity.model.MatchStrength;
+import org.commcare.commcaresupportlibrary.identity.model.RegistrationResult;
+import org.commcare.commcaresupportlibrary.identity.model.VerificationMatch;
+import org.commcare.util.LogTypes;
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.AbstractTreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.core.services.Logger;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.core.util.OrderedHashtable;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Vector;
+
+import androidx.annotation.StringDef;
+
+public class IdentityCalloutHandler {
+
+
+    private static final String REF_GUID = "guid";
+    private static final String REF_CONFIDENCE = "confidence";
+    private static final String REF_MATCH_STRENGTH = "match_strength";
+
+    public static final String GENERALIZED_IDENTITY_PROVIDER = "generalized_identity_provider";
+
+
+    @StringDef({GENERALIZED_IDENTITY_PROVIDER, SimprintsCalloutProcessing.SIMPRINTS_IDENTITY_PROVIDER})
+    @Retention(RetentionPolicy.SOURCE)
+    public @interface IdentityProvider {
```

Comment:
Is there still some way to specifically choose each identity provider (with, say, an ID the same way that SimPrints works?)?

I doubt there will often be more than one in use at a time, but I imagine inside of an app, when people configure "Their identify provider" they may want to choose a specific one.

Reply:
Choice of Identity provider totally depends on the callout action the app defines i.e. "org.simprint.action" vs something else. The Identity provider scpecific handling in the app only applies on the Intent reponse handling which we do by checking the format of the response in intent like [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/provider/SimprintsCalloutProcessing.java#L41).